LLama/Experimental/Abstractions/ISamplingMethod.cs

```diff
@@ -0,0 +1,44 @@
+﻿using LLama.Experimental.Common;
+using LLama.Experimental.Runner.LLamaCpp;
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace LLama.Experimental.Abstractions
+{
+    /// <summary>
+    /// Method to sample the model output.
+    /// </summary>
+    public interface ISamplingMethod
```

Comment:
This looks similar to `ISamplingPipeline`, is it basically that?

Reply:
Yes, I renamed it because I was not sure if there'll be slightly different from it. Using a different name will not introduce modifications in non-experimental part.